spring/src/main/java/io/seata/spring/util/GlobalTransactionalCheck.java

```diff
@@ -0,0 +1,134 @@
+    private static volatile boolean disable;
+    private static ScheduledThreadPoolExecutor executor =
+        new ScheduledThreadPoolExecutor(1, new NamedThreadFactory("degradeCheckWorker", 1, true));
+
+    public GlobalTransactionalCheck() {
+        disable = ConfigurationFactory.getInstance().getBoolean(ConfigurationKeys.DISABLE_GLOBAL_TRANSACTION,
+            DEFAULT_DISABLE_GLOBAL_TRANSACTION);
+        degradeCheck = ConfigurationFactory.getInstance().getBoolean(ConfigurationKeys.CLIENT_DEGRADE_CHECK,
+            DEFAULT_TM_DEGRADE_CHECK);
+        if (degradeCheck) {
+            degradeCheckPeriod = ConfigurationFactory.getInstance()
+                .getInt(ConfigurationKeys.CLIENT_DEGRADE_CHECK_PERIOD, DEFAULT_TM_DEGRADE_CHECK_PERIOD);
+            degradeCheckAllowTimes = ConfigurationFactory.getInstance()
+                .getInt(ConfigurationKeys.CLIENT_DEGRADE_CHECK_ALLOW_TIMES, DEFAULT_TM_DEGRADE_CHECK_ALLOW_TIMES);
+            if (degradeCheckPeriod > 0 && degradeCheckAllowTimes > 0) {
+                startDegradeCheck();
+            }
+        }
+    }
+
+    /**
+     * auto upgrade service detection
+     */
+    private static void startDegradeCheck() {
+        executor.scheduleAtFixedRate(() -> {
+            if (degradeCheck) {
+                try {
+                    String xid = TransactionManagerHolder.get().begin(null, null, "degradeCheck", 60000);
+                    TransactionManagerHolder.get().commit(xid);
+                    onDegradeCheck(true);
+                } catch (Exception e) {
+                    onDegradeCheck(false);
+                }
+            }
+        }, degradeCheckPeriod, degradeCheckPeriod, TimeUnit.MILLISECONDS);
+    }
+
+    public static synchronized void onDegradeCheck(boolean succeed) {
+        if (succeed) {
+            if (degradeNum >= degradeCheckAllowTimes) {
+                reachNum++;
+                if (reachNum >= degradeCheckAllowTimes) {
+                    reachNum = 0;
+                    degradeNum = 0;
+                    if (LOGGER.isInfoEnabled()) {
+                        LOGGER.info("the current global transaction has been restored");
+                    }
+                }
+            } else if (degradeNum != 0) {
+                degradeNum = 0;
+            }
+        } else {
+            if (degradeNum < degradeCheckAllowTimes) {
+                degradeNum++;
+                if (degradeNum >= degradeCheckAllowTimes) {
+                    if (LOGGER.isWarnEnabled()) {
+                        LOGGER.warn("the current global transaction has been automatically downgraded");
+                    }
+                }
+            } else if (reachNum != 0) {
+                reachNum = 0;
+            }
+        }
+    }
+
+    public static boolean getStatus() {
+        if (!degradeCheck) {
+            return disable;
+        }
+        return disable || (degradeCheck && degradeNum >= degradeCheckAllowTimes);
```

Comment:
I think degradeCheck always true in here.

Reply:
PTAL